deeppavlov/models/embedders/rove_embedder.py

```diff
@@ -0,0 +1,262 @@
+                matrix = tf.matmul(seq_context, seq_context, transpose_b=True)
+                loss2 += 1. - matrix
+
+        self.target = final_vectors[-1]
+        self.cost = tf.reduce_sum(loss1) / self.batch_size / self.seq_length
+        self.cost += tf.reduce_sum(loss2) / self.batch_size / self.seq_length
+        self.final_state = last_state
+        self.lr = tf.Variable(0.0, trainable=False)
+        tvars = tf.trainable_variables()
+
+        with tf.variable_scope(tf.get_variable_scope(), reuse=False):
+            grads, _ = tf.clip_by_global_norm(tf.gradients(self.cost, tvars),
+                                                  self.grad_clip)
+            self._optimizer = tf.train.AdamOptimizer(self.lr)
+            self.train_op = self._optimizer.apply_gradients(zip(grads, tvars))
+
+        self.sess = tf.Session(config=config)
+        tf.global_variables_initializer().run(session=self.sess)
+        self._saver = tf.train.Saver(tf.global_variables())
+
+    def _set_state(self, state):
+        self.initial_state = state
+
+    def _train_step(self, batch, change):
+        if self.initial_state is None:
+            state = self.initial_state
+        else:
+            state = self.initial_state.eval(session=self.sess)
+
+        feed = {self.input_data: batch,
+                self.change: change,
+                self.initial_state: state}
+        train_loss, state, _ = self.sess.run([self.cost, self.final_state, self.train_op], feed)
+        self._set_state(state)
+        return train_loss
+
+    def _forward(self, batch, change):
+        if self.initial_state is None:
+            state = self.initial_state
+        else:
+            state = self.initial_state.eval(session=self.sess)
+
+        feed = {self.input_data: batch,
+                self.change: change,
+                self.initial_state: state}
+        train_loss, state, _ = self.sess.run([self.cost, self.final_state, self.train_op], feed)
+        self._set_state(state)
+        return train_loss
+
+    def _encode(self, sentence: str):
+        state = self.cell.zero_state(1, tf.float32).eval(session=self.sess)
+
+        tokens = word_tokenize(sentence)
+        vectors = []
+        for token in tokens:
+            x = self._letters2vec(token, self.vocab).reshape((1, 1, -1))
+            feed = {self.input_data: x,
+                    self.initial_state: state,
+                    self.change: np.zeros((1,))
+                    }
+            [state, target] = self.sess.run([self.final_state, self.target], feed)
+            vectors.append(np.squeeze(target))
+
+        embs = np.array(vectors)
+        return embs
+
+    @check_attr_true('train_now')
+    def train(self, data_loader, **kwargs):
+        print(':: creating new  model\n')
+        self._train_step(data_loader)
+        self.save()
+
+    def infer(self, sentence: str, *args, **kwargs):
+        return self._encode(sentence)
+
+    def load(self):
```

Comment:
5. Why `TFModel.save()` and `TFModel.load()` are not used here? If they are no good for this usecase, this is a serious lack of flexibility in the abstract class and the abstract class methods should be rewritten.